packages/Accordion/webpack.config.js

```diff
@@ -0,0 +1,76 @@
+const path = require("path");
+const ForkTsCheckerWebpackPlugin = require("fork-ts-checker-webpack-plugin");
+const ESLintPlugin = require("eslint-webpack-plugin");
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');
+
+module.exports = {
+  mode: "production",
+  entry: "./src/index.ts",
+  output: {
+    path: path.resolve(__dirname, "dist"),
+    filename: "[name].[contenthash].js",
+    clean: true
+  },
+  optimization: {
+    minimize: true,
+    minimizer: [`...`, new CssMinimizerPlugin()],
+    usedExports: true,
+    removeAvailableModules: true,
+    concatenateModules: true
+  },
+  resolve: {
+    extensions: [".js", ".json", ".ts", ".tsx"],
+  },
+  module: {
+    rules: [
+      {
+        test: /\.(ts|js)x?$/i,
+        exclude: /node_modules/,
+        use: {
+          loader: "babel-loader",
+          options: {
+            presets: [
+              "@babel/preset-env",
+              "@babel/preset-react",
+              "@babel/preset-typescript",
+            ],
+          },
+        },
+      },
+      {
+        test: /\.scss$/,
+        use: [
+          MiniCssExtractPlugin.loader,
+          "css-loader",
+          {
+            loader: "postcss-loader",
+            options: {
+              postcssOptions: {
+                plugins: [
+                  ["postcss-preset-env"],
+                ],
+              },
+            },
+          },
+          "sass-loader"
+        ],
+      }],
+  },
+  externals: {
```

Comment:
No, `externals` is setting `react` and `react-dom` as external dependencies so they are not bundled.

Reply:
This comment is still something I'm not sure about as I'm not sure of the benefits of bundling dependencies at publish rather then when the app is built as I'm not sure if webpack will be able to deduplicate and tree-shake as well but we can test in prerelease branch and see

Reply:
Before going into more details just to clarify that we are not bundling dependencies by using `externals` option of webpack, but quite the opposite - we are preventing webpack not to bundle those deps such as `react`

The way the library was built initially as a monorepo making each component completely independent of each other has some advantages such as you can update a component and release a new version of that component instead of releasing the entire library but it has some caveats as dependencies are duplicated and bundled.

If we would like to remove the duplicated dependencies we would need not to organise the packages in a monorepo so instead of having @blaze-react/select we would have only @blaze-react and imports would be named imports e.g import { Select } from '@blaze-react'; that would work perfectly for webpack and tree-shaking as Amazon is doing with @aws-sdk and so on...

There's a workaround though that can work with how we have had things organised in a monorepo and it is to declare external dependencies. For example, we have a library of react components and those components are going to be hosted by a react or next.js app, so you can say webpack not to bundle react/prop-types/react-dom from the bundle but it won't be as effective as not using a monorepo for it.

So far we have these,

Blaze-admin (react app and host)
  -- React
       -- Blaze React Components

Frontend ( next.js app which includes react, react-dom and so on)
  -- React
       -- Blaze React Components

So webpack externals option is not bundling react inside each component making it be smaller.

I don't know the architectural decision made behind using a monorepo though but maybe is something we should improve in the near future, though.

Let me know please if that doesn't reply to your questions or if anything is not clear or if I am missing something and we can discuss it.

Reply:
I'm still not fully sure how webpack will tree shake some bits and I've put some comments below but I think once we have a prerelease process setup it will be worth moving forward and publishing as it is now so we can try on a branch in blaze and I can see how it works in practice I'll be able to see and tweak if we need to as its much harder for you to test and develop without publishing 👍 

With regards to the external dependencies sorry didn't explain very well, I can see react etc are marked as externals so won't be bundled but other dependencies will be bundled on published e.g. `lodash.differencewith`, `react-datepicker`, `draft-js-export-html`. 

When used in blaze where the data-ui plugin also uses `lodash.differencewith` then I thought the admin bundle would end up having two copies of the code as it would have the copy bundled in `@blaze-react/multiselect` and the dependency from the data-ui plugin? 

With regards to the use of the mono repo it shouldn't be duplicating any dependencies as they don't get bundled on publish so will only get installed where they are used (e.g. in blaze/project) and then npm/yarn installs the required version which should be the same as in the projects so there is only one version/copy of `react` installed.

But having a single module that bundles them and all can be tree-shaken could be a simpler way of keeping the components up to date an making them simpler to manage, definitely something we should look into 👍